instrumentation/oshi/library/src/test/java/io/opentelemetry/instrumentation/oshi/SystemMetricsTest.java

```diff
@@ -16,12 +19,18 @@ class SystemMetricsTest extends AbstractSystemMetricsTest {
   public static final InstrumentationExtension testing = LibraryInstrumentationExtension.create();
 
   @Override
-  protected void registerMetrics() {
-    SystemMetrics.registerObservers(GlobalOpenTelemetry.get());
+  protected List<AutoCloseable> registerMetrics() {
+    return SystemMetrics.registerObservers(GlobalOpenTelemetry.get());
   }
 
   @Override
   protected InstrumentationExtension testing() {
     return testing;
   }
+
+  @Test
+  void closeObservables() {
```

Comment:
`closeObservables` isn't a good name for the test because it doesn't really close anything, you just verify that the list isn't empty
doing it like this isn't ideal because the test in `AbstractSystemMetricsTest` doesn't close the observables you'll end up with a bunch of warnings like `WARNING: Instrument system.memory.usage has recorded multiple values for the same attributes: {state="used"}` when you register the metrics twice. You could register the metrics once with
```
  private static List<AutoCloseable> metrics;

  @BeforeAll
  static void setUp() {
    metrics = SystemMetrics.registerObservers(GlobalOpenTelemetry.get());
  }
```
and change `registerMetrics` to
```
  @Override
  protected void registerMetrics() {}
```
or completely remove it

Reply:
Fair point. I updated tests in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10364/commits/dca9f658e03aba79e2df590daac4bdfa7f4e7a7e.